src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs

```diff
@@ -197,11 +199,140 @@ private void CheckStreamIsProcessable(Stream stream)
             }
         }
 
+        //Could put FileMissingLars, FileInvalidLars and FileExpiredLars into one method but they may come in useful elsewhere
```

Comment:
I think these checks should be in a single method - `internal async Task<(string[] Invalid, string[] Expired)> CheckLearnAimRefs(Stream stream);`

What's the difference between invalid and missing?

Also the implementation can parse to `CsvCourseRow` instead of dynamic.

Reply:
I asked @MrKevJoy if I should put them into a single method and he said keep them seperate :-) I happy to change if everyone agrees.

Missing is when a code doesn't exist, Invalid when it's not in the table. I'm going with what errors we have in the prototype.

Reply:
Missing/invalid makes sense.

Given the implementations are almost identical and they all call a (potentially expensive) SQL call I think it should be one method with one DB call.

Reply:
Do you want me to keep the other methods in or strip them out?

Reply:
Ideally `FileMissingLars`, `FileInvalidLars`, `FileExpiredLars` are all replaced with `CheckLearnAimRefs`